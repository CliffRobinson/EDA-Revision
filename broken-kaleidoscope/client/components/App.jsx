import React from 'react'

import Pixel from './Pixel'

export default class App extends React.Component {
  constructor (props) {
    super(props)

    let butts = []

    for (let i = 0; i < 1500; i++) {
      butts.push({
        color1: randomHexColor(),
        color2: randomHexColor(),
        angle: randomAngle()
      })
    }

    this.state = {
      butts: butts
    }

  }

  componentDidMount() {
    setInterval(
      () => {
        //console.log("refreshing")
        const newButts = this.state.butts.slice()
        const newPixel = {
          color1: randomHexColor(),
          color2: randomHexColor(),
          angle: randomAngle()
        }
        const randIndex = Math.floor(Math.random() * newButts.length)
        newButts[randIndex] = newPixel
        this.setState({
          butts:newButts
        }) 
      }, 1
    )
  }

  render () {
    //console.log("rendering")
    const { butts } = this.state
    return butts.map(butt => (
      <Pixel color1={butt.color1} color2={butt.color2} angle={butt.angle} />
    ))
  }
}

const randomHexColor = () =>
  `#${Math.floor(Math.random() * 0x1000000).toString(16).padStart(6, 0)}`

const randomAngle = () => `${Math.floor(Math.random() * 360)}deg`
