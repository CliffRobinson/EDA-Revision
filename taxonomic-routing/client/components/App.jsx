import React, { Fragment } from 'react'
import { HashRouter as Router, Route } from 'react-router-dom'

import Nav from './Nav'
import Home from './Home'
import Rank from "./Rank"
import Classification from './Classification';
import SpeciesListing from './SpeciesListing'

const App = () => {
  return (
    <Router>
      <div className='app'>
        <h1>Welcome to Tax Town!</h1>
        <h3>Don't worry Objectivists, it's Taxonomy, not Taxation</h3>
        <Nav />
        <Route exact path="/" component={Home} />
        <Route path="/list/:rank" component={Rank}/>
        <Route path="/rank/:rank/:classification" component={Classification}/>
        <Route path="/rank/:rank/:classification/species" component={SpeciesListing}/>
        <Route exact path="/test" render={
          () => <p >lol</p>
        }/>
      </div>
    </Router>
  )
}

export default App



