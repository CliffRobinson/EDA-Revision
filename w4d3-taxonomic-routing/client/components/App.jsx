import React from 'react'
import { HashRouter as Router, Route } from 'react-router-dom'

import { Nav } from './Nav'
import { Home } from './Home'
import { Classifications} from './Classifications'
import { Rank } from './Rank'
import {Species} from './Species'

import ranks from '../../data/ranks'
import specieses from '../../data/species'

export const capitaliseFirst = (str) => (str[0].toUpperCase()+str.substring(1))

const App = (props) => {

  return (
    <Router>
      <React.Fragment>
        <h1> HERE IS THE HEADING !!</h1>
        <div style={{ display: "flex" }}>
          <div style={{ width: "250px" }}><Nav /></div>
          <div>
            <Route exact path="/" component={Home} />
            {Object.keys(ranks)
              .map((classificationName, i) => (<Route key={i} exact path={`/list/${classificationName}`}  
              render={()=> <Classifications name={capitaliseFirst(classificationName)} 
              classifications={ranks[classificationName]} />}/>))
            }
            {Object.keys(ranks).map(
              (classificationName) => ranks[classificationName].map(
                (rank, j) => {
                return <Route key={classificationName+j} exact path={`/${classificationName}/${rank.name}`} 
                render={()=> <Rank classification={classificationName} rank={rank}/>} 
                />
                }
              )
            )}
            {specieses.map(
              (specie, k) => {
              return <Route key={"specie"+k} exact path={`/species/${specie.id}`} 
              render={()=><Species {... specie}
              />} />
              }
            )}
          </div>
        </div>
      </React.Fragment>
    </Router>

  )
}

export default App

