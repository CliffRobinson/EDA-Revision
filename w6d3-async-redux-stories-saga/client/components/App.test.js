import React from 'react'
import {shallow} from 'enzyme'

import App from './App'

test('App.jest', ()=> {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
})