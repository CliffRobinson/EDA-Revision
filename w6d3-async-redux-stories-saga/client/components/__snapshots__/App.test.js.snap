// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.jest 1`] = `
<div
  className="app"
>
  <Connect(ErrorMessage) />
  <Connect(LoadSubreddit)>
    <Connect(WaitIndicator) />
  </Connect(LoadSubreddit)>
  <Connect(Subreddit) />
</div>
`;
